### ISSUE TITLE
Compilation failure with newest dependencies

### ISSUE THREAD

**Issue by snoyberg** [athanclark/pred-trie#2]
```
[3 of 4] Compiling Data.Trie.Pred.Unified.Tail ( src/Data/Trie/Pred/Unified/Tail.hs, dist/build/Data/Trie/Pred/Unified/Tail.o )

src/Data/Trie/Pred/Unified/Tail.hs:55:23:
    Couldn't match expected type ‘a7 -> b8 -> c7’
                with actual type ‘UPTrie t (r -> x)’
    Relevant bindings include
      xrs :: UPTrie t (r -> x)
        (bound at src/Data/Trie/Pred/Unified/Tail.hs:55:16)
      t :: t (bound at src/Data/Trie/Pred/Unified/Tail.hs:55:14)
      p :: t -> Maybe r
        (bound at src/Data/Trie/Pred/Unified/Tail.hs:55:12)
      suppliment :: (t -> Maybe r)
                    -> t -> UPTrie t (r -> x) -> Maybe (UPTrie t x)
        (bound at src/Data/Trie/Pred/Unified/Tail.hs:55:1)
    In the first argument of ‘(<~$>)’, namely ‘xrs’
    In the first argument of ‘(<$>)’, namely ‘(xrs <~$>)’

src/Data/Trie/Pred/Unified/Tail.hs:96:72:
    Couldn't match type ‘UPTrie t1 (r -> a9 -> c)’ with ‘a9 -> b9 -> c’
    Expected type: [a9 -> b9 -> c]
      Actual type: [UPTrie t1 (r -> a9 -> c)]
    Relevant bindings include
      xrs :: [UPTrie t1 (r -> a9 -> c)]
        (bound at src/Data/Trie/Pred/Unified/Tail.hs:95:25)
      mrx :: Maybe (r -> a9 -> c)
        (bound at src/Data/Trie/Pred/Unified/Tail.hs:95:21)
      p :: t1 -> Maybe r
        (bound at src/Data/Trie/Pred/Unified/Tail.hs:95:19)
      t :: t1 (bound at src/Data/Trie/Pred/Unified/Tail.hs:95:17)
      go :: UPTrie t1 (a9 -> c) -> WrappedMonoid (Maybe (a9 -> c))
        (bound at src/Data/Trie/Pred/Unified/Tail.hs:94:7)
    In the second argument of ‘mapMaybe’, namely ‘xrs’
    In the first argument of ‘mconcat’, namely
      ‘(mapMaybe
          (\ z -> (\ r -> unwrapMonoid $ go $ z <~$> r) <$> p t) xrs)’
```

**Comment by athanclark** [comment]
This was a faulty version change on my part - composition-extra should be
at version 0.2.0, and the constraint in pred-trie should reflect this. I
will deprecate and update shortly

**Comment by athanclark** [comment]
I patched the cabal file for pred-trie-0.2.4 on hackage to reflect the
dependence on outdated terms. There will be large breaking changes soon for
both packages, and these versions will be deprecated.
